File: client/src/app/wto/appellatio-comis.enum.ts
/* @tt{{{


    @move-comment-backward
    @template-renderer [
        templateRendererClassName="EnumDefinitionTypescriptRenderer"
        templateRendererPackageName="senegai.codegen.renderer.angular"
        templateRendererInterfaceName="UiEnumRenderer"
        templateRendererInterfacePackageName="senegai.codegen.renderer.angular"
    ] [
        modelClassName="UiEnumModel"
        modelPackageName="senegai.codegen.renderer.model.ui"
        modelName="model"
    ]

    @replace-value-by-expression
        [ searchValue="AppellatioComis" replaceByExpression="model.enumName.pascalCase" ]
        [ searchValue="appellatioComis" replaceByExpression="model.enumName.camelCase" ]
        [ searchValue="appellatio-comis" replaceByExpression="model.enumName.kebabCase" ]

    @modify-provided-filename-by-replacements


}}}@ */
export enum AppellatioComisEnum {
    /* @tt{{{
        @foreach [ iteratorExpression="model.enumValues" loopVariable="enumValue" ]

        @replace-value-by-expression
            [ searchValue="VIR_HONORATUS" replaceByExpression="enumValue.screamingSnakeCase" ]

    }}}@  */
    VIR_HONORATUS = 'VIR_HONORATUS',
    /* @tt{{{   @end-foreach  }}}@ */
    /* @tt{{{   @ignore-text  }}}@ */
    FEMINA_HONESTA = 'FEMINA_HONESTA',
    /* @tt{{{   @end-ignore-text  }}}@ */
}

export const AppellatioComisEnumValues: ReadonlyArray<AppellatioComisEnum> = [
    /* @tt{{{
        @foreach [ iteratorExpression="model.enumValues" loopVariable="enumValue" ]

        @replace-value-by-expression
            [ searchValue="VIR_HONORATUS" replaceByExpression="enumValue.screamingSnakeCase" ]

    }}}@  */
    AppellatioComisEnum.VIR_HONORATUS,
    /* @tt{{{   @end-foreach  }}}@ */
    /* @tt{{{   @ignore-text  }}}@ */
    AppellatioComisEnum.FEMINA_HONESTA,
    /* @tt{{{   @end-ignore-text  }}}@ */
]

File: server/rest/src/main/kotlin/senegai/server/restapi/wto/AppellatioComisEnum.kt
/* @tt{{{

    @move-comment-backward
    @template-renderer [
        templateRendererClassName="EnumWTORenderer"
        templateRendererPackageName="senegai.codegen.renderer.be"
        templateRendererInterfaceName="BeEnumRenderer"
        templateRendererInterfacePackageName="senegai.codegen.renderer.be"
    ] [
        modelClassName="BeEnumModel"
        modelPackageName="senegai.codegen.renderer.model.be"
        modelName="model"
    ]

    @replace-value-by-expression
        [ searchValue="AppellatioComis" replaceByExpression="model.enumName.pascalCase" ]

    @modify-provided-filepath-by-replacements

}}}@ */
package senegai.server.restapi.wto

/**
 * WTO (Web Transfer Object) enum mirroring the Angular `AppellatioComisEnum`.
 */
enum class AppellatioComisEnum {
    /* @tt{{{
        @foreach [ iteratorExpression="model.enumValues" loopVariable="enumValue" ]
        @replace-value-by-expression
            [ searchValue="VIR_HONORATUS" replaceByExpression="enumValue.screamingSnakeCase" ]
    }}}@ */
    VIR_HONORATUS,
    /* @tt{{{   @end-foreach  }}}@ */
    /* @tt{{{   @ignore-text  }}}@ */
    FEMINA_HONESTA,
    /* @tt{{{   @end-ignore-text  }}}@ */
}
